src/flow/io/event.js

```diff
@@ -358,9 +358,15 @@ function applyCalendarDefaults(calendar, user) {
 }
 
 function fetchAndParseIcal(src) {
-  return fetch(src)
-    .then(result => result.text())
-    .then(iCalParseEvents)
+  if (src.startsWith('http')) {
+    return fetch(src)
+      .then(result => result.text())
+      .then(iCalParseEvents)
+  }
+
+  return new Promise(resolve => {
```

Comment:
Could be `Promise.resolve(iCalParseEvents(src))`.
However, the if should be made outside the fetch and parse ical method call. Or it should be a parameter on the method. This mixes the meaning of the parameters too much.

Reply:
Done.